nncf/common/pruning/symbolic_mask.py

```diff
@@ -27,17 +58,22 @@ class SymbolicMask(NNCFTensor):
     symbolic mask propagation by SymbolicMaskProcessor.
     """
 
-    def __init__(self, dimension: int, mask_producers: List[int] = None):
+    def __init__(self, dimension: int, mask_producers: Union[int, List[SymbolicMaskProducer]] = None):
```

Comment:
In which cases are mask producers equal to int?

Reply:
Im `symbolic_mask_propagation`, just sugar to not initialize symbolic mask like
```
mask = SymbolicMask(dimension=dim, mask_producers=MaskProducer(producer))
```

it works like that instead:
```
mask = SymbolicMask(dimension=dim, mask_producers=producer)
```